### PR TITLE
refactor nextcloud.env file creation

### LINKED ISSUE #2: Why do you delete and recreate the env file?
Thanks for the nice setup, Christian!

I need to dig into the details, as you set a lot of options, but this far it looks very slick and promising.

Just one question, out of curiosity:

Why to you remove the `nextcloud.env` file and then create it line by line from scratch?
https://github.com/LorbusChris/nextcloud-quadlet/blob/f8028e75b2bd4b53f07172c9e7a516419ebe91f3/playbook.yml#L96

Replacing the remove, touch and lineinfile tasks (lines 96 to 147) with just one task using a Jinja template using the same variables would do the same job?

Unless I am overlooking some cornercase, hence my question.

Kind Regards
Johannes

### PR DESCRIPTION
**Pull request by johanneskastl**
fix #2 

I can try to test this tomorrow, but ansible-lint is happy already...

### CODE REVIEW

**Comment by LorbusChris** [comment]
Neat, thanks! LGTM, too. I'll let you test and report back here before I hit the merge button!

File: playbook.yml [hunk 1]
```diff
@@ -93,59 +93,12 @@
...
-      path: ~/storage/nextcloud/nextcloud.env
-      state: absent
-
-  - name: Touch EnvironmentFile
-    ansible.builtin.file:
-      path: ~/storage/nextcloud/nextcloud.env
-      state: touch
+  - name: Create EnvironmentFile
+    ansible.builtin.template:
+      src: nextcloud.env.j2
```
Comment (LorbusChris) [inline playbook.yml:98]:
```suggestion
      src: templates/nextcloud.env.j2
```
Or is the `templates` dir the lookup dir anyway?

Comment (johanneskastl) [reply playbook.yml:98]: Yes, `templates` is the default location, that is searched first.

https://docs.ansible.com/ansible/latest/playbook_guide/playbook_pathing.html

**Comment by kastl-ars** [comment]
I tested the change successfully. A file, create by a playbook run on the main branch, was shown as "OK" (no changes needed) on this branch.
A file newly created by a playbook run on this branch was identical to the one created on the main branch.

**Comment by LorbusChris** [comment]
:shipit:

**Pull request merged**